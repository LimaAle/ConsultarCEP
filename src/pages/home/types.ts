export interface HomeViewModel{
    cep:string;
    setCep:React.Dispatch<React.SetStateAction<string>>;
    isLoading:boolean;
    onSubmit:()=>void;
}