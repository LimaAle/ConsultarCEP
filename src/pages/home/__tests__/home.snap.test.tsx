import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { HomeViewModel } from "../types";
import HomeView from '../view';
import useHomeViewModel from '../view.model';

jest.mock('../view.model', () => {
  const fnHomeViewModel = jest.fn();
  fnHomeViewModel.mockImplementation(() => ({
    cep: '',
    setCep: jest.fn(),
    isLoading: false,
    onSubmit: jest.fn(),
    hasError:false,
    errorMessage:'error',
    clearText:jest.fn(),
    noInternetMessage:'nomessage',
    hasInternet:false
  } as HomeViewModel));
  return fnHomeViewModel;
});
describe('home view tests',()=>{
  it('should render', () => {
    const { toJSON } = renderer.create(<HomeView />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('should render with cep', () => {
    useHomeViewModel.mockImplementation(() => ({
      cep: '04654000',
      setCep: jest.fn(),
      isLoading: false,
      onSubmit: jest.fn()
    }))
  
    const { toJSON } = renderer.create(<HomeView />);
    expect(toJSON()).toMatchSnapshot();
  })
})


