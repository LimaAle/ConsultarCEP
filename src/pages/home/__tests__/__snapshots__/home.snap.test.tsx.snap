// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home view tests should render 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    {
      "colors": [
        4280490306,
        4284372110,
        4288644800,
        4293189594,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#eee",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTSafeAreaView
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "#eee",
            "fontSize": 24,
          }
        }
      >
        Digite um CEP para pesquisar.
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ddd",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "margin": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <Text />
        <TextInput
          keyboardType="numeric"
          maxLength={9}
          onChangeText={[Function]}
          placeholder="CEP"
          style={
            {
              "flex": 1,
              "fontStyle": undefined,
              "fontWeight": undefined,
              "marginLeft": 15,
              "maxWidth": 150,
              "textDecorationLine": undefined,
            }
          }
          value=""
        />
        <View
          accessibilityState={
            {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text />
        </View>
      </View>
      <Text
        style={
          {
            "color": "#b7094c",
            "fontSize": 18,
            "marginVertical": 8,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#231942",
            "borderRadius": 20,
            "elevation": 4,
            "opacity": 1,
            "padding": 10,
            "paddingHorizontal": 15,
            "shadowOffset": {
              "height": 4,
              "width": -2,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 3,
          }
        }
      >
        <Text
          style={
            {
              "color": "#eee",
            }
          }
        >
          PESQUISAR
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#b7094c",
            "marginVertical": 12,
            "padding": 8,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#eee",
            }
          }
        >
          Sem internet!
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</ViewManagerAdapter_ExpoLinearGradient>
`;

exports[`home view tests should render with cep 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    {
      "colors": [
        4280490306,
        4284372110,
        4288644800,
        4293189594,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#eee",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTSafeAreaView
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          {
            "color": "#eee",
            "fontSize": 24,
          }
        }
      >
        Digite um CEP para pesquisar.
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ddd",
            "borderRadius": 20,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "margin": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <Text />
        <TextInput
          keyboardType="numeric"
          maxLength={9}
          onChangeText={[Function]}
          placeholder="CEP"
          style={
            {
              "flex": 1,
              "fontStyle": undefined,
              "fontWeight": undefined,
              "marginLeft": 15,
              "maxWidth": 150,
              "textDecorationLine": undefined,
            }
          }
          value=""
        />
        <View
          accessibilityState={
            {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text />
        </View>
      </View>
      <Text
        style={
          {
            "color": "#b7094c",
            "fontSize": 18,
            "marginVertical": 8,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#231942",
            "borderRadius": 20,
            "elevation": 4,
            "opacity": 1,
            "padding": 10,
            "paddingHorizontal": 15,
            "shadowOffset": {
              "height": 4,
              "width": -2,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 3,
          }
        }
      >
        <Text
          style={
            {
              "color": "#eee",
            }
          }
        >
          PESQUISAR
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#b7094c",
            "marginVertical": 12,
            "padding": 8,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#eee",
            }
          }
        >
          Sem internet!
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</ViewManagerAdapter_ExpoLinearGradient>
`;
